api/routers/legacy.py

```diff
@@ -1217,31 +1217,33 @@ async def get_heatmap_data(token: str, region_id: RegionID):
 
 
 @router.post('/discord/player_bans/{token}', tags=["Legacy"])
-async def generate_excel_export(token: str, export_info: ExportInfo):
+async def generate_excel_export(token: str, export_info: ExportInfo, tasks: BackgroundTasks):
+    '''Initiates a task to generate an excel or csv bans export for a user.'''
+
     await verify_token(token, verification='verify_players')
-    #get_ban_spreadsheet_data
 
     req_data = export_info.dict()
     discord_id = req_data.get('discord_id')
     display_name = req_data.get('display_name')
-    file_type =req_data.get('file_type')
+    file_type = req_data.get('file_type')
+
+    if file_type not in ["excel", "csv"]:
+        raise HTTPException(status_code=400, detail="File type specified is invalid.")
 
     linked_accounts = await sql_get_discord_linked_accounts(discord_id)
 
     if len(linked_accounts) == 0:
         raise HTTPException(status_code=500, detail="User doesn't have any accounts linked.")
```

Comment:
500 indicates server error, however this is a user error.
for more verbose status code:
`fastapi.status.<click>`